content/actions/using-workflows/workflow-commands-for-github-actions.md

```diff
@@ -412,6 +408,191 @@ jobs:
+        shell: pwsh
+        run: |
+          Set-Variable -Name TheSecret -Value (Get-Random)
+          Write-Output "::add-mask::$TheSecret"
+{% ifversion actions-save-state-set-output-envs %}
+          "secret-number=$TheSecret" >> $env:GITHUB_OUTPUT
+{% else %}
+          Write-Output "::set-output name=secret-number::$TheSecret"
+{% endif %}
+      - name: Use that secret output (protected by a mask)
+        shell: pwsh
+        run: |
+{% raw %}
+          Write-Output "the secret number is ${{ steps.sets-a-secret.outputs.secret-number }}"
+{% endraw %}
+```
+
+{% endpowershell %}
+
+### Example: Masking and passing a secret between jobs{% ifversion fpt or ghec or ghes > 3.3 or ghae > 3.3 %} or workflows{% endif %}
+
+If you want to pass a masked secret between jobs{% ifversion fpt or ghec or ghes > 3.3 or ghae > 3.3 %} or workflows{% endif %}, you should store the secret in a store and then retrieve it in the subsequent job{% ifversion fpt or ghec or ghes > 3.3 or ghae > 3.3 %} or workflow{% endif %}.
+
+#### Setup
+1. Set up a secret store to store the secret that you will generate during your workflow. For example, Vault.
+1. Generate a key for reading and writing to that secret store. Store the key as a repository secret. In the following example workflow, the secret name is `SECRET_STORE_CREDENTIALS`. For more information, see "[Encrypted secrets](/actions/security-guides/encrypted-secrets)."
+
+#### Workflow
+
+{% note %}
+
+**Note**: This workflow uses an imaginary secret store, `secret-store`, which has imaginary commands `store-secret` and `retrieve-secret`. `some/secret-store@ 27b31702a0e7fc50959f5ad993c78deac1bdfc29` is an imaginary action that installs the `secret-store` application and configures it to connect to an `instance` with `credentials`.
+
+{% endnote %}
+
+{% bash %}
+
+```yaml{:copy}
+on: push
+
+jobs:
+  secret-generator:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: some/secret-store@v1
+      with:
+{% raw %}
+        credentials: ${{ secrets.SECRET_STORE_CREDENTIALS }}
+        instance: ${{ secrets.SECRET_STORE_INSTANCE }}
+{% endraw %}
+    - name: generate secret
+      shell: bash
+      run: |
+        GENERATED_SECRET=$((RANDOM))
+        echo "::add-mask::$GENERATED_SECRET"
+        SECRET_HANDLE=$(secret-store store-secret "$GENERATED_SECRET")
+{% ifversion actions-save-state-set-output-envs %}
+        echo "handle=$secret_handle" >> "$GITHUB_OUTPUT"
+{% else %}
+        echo "::set-output name=handle::$secret_handle"
+{% endif %}
+  secret-consumer:
+    runs-on: macos-latest
+    needs: secret-generator
+    steps:
+    - uses: some/secret-store@v1
+      with:
+{% raw %}
+        credentials: ${{ secrets.SECRET_STORE_CREDENTIALS }}
+        instance: ${{ secrets.SECRET_STORE_INSTANCE }}
+{% endraw %}
+    - name: use secret
+      shell: bash
+      run: |
+{% raw %}
+        SECRET_HANDLE="${{ needs.secret-generator.outputs.handle }}"
+{% endraw %}
```

Comment:
Move tags to avoid blank lines in output:

```suggestion
      with:{% raw %}
        credentials: ${{ secrets.SECRET_STORE_CREDENTIALS }}
        instance: ${{ secrets.SECRET_STORE_INSTANCE }}{% endraw %}
    - name: generate secret
      shell: bash
      run: |
        GENERATED_SECRET=$((RANDOM))
        echo "::add-mask::$GENERATED_SECRET"
        SECRET_HANDLE=$(secret-store store-secret "$GENERATED_SECRET"){% ifversion actions-save-state-set-output-envs %}
        echo "handle=$secret_handle" >> "$GITHUB_OUTPUT"{% else %}
        echo "::set-output name=handle::$secret_handle"{% endif %}
  secret-consumer:
    runs-on: macos-latest
    needs: secret-generator
    steps:
    - uses: some/secret-store@v1
      with:{% raw %}
        credentials: ${{ secrets.SECRET_STORE_CREDENTIALS }}
        instance: ${{ secrets.SECRET_STORE_INSTANCE }}{% endraw %}
    - name: use secret
      shell: bash
      run: |{% raw %}
        SECRET_HANDLE="${{ needs.secret-generator.outputs.handle }}"{% endraw %}
```